packages/core/src/tracing/dynamicSamplingContext.ts

```diff
@@ -41,12 +41,21 @@ export function getDynamicSamplingContextFromClient(trace_id: string, client: Cl
 
   const { publicKey: public_key } = client.getDsn() || {};
 
-  const dsc = dropUndefinedKeys({
+  // Instead of conditionally adding non-undefined values, we add them and then remove them if needed
+  // otherwise, the order of baggage entries changes, which "breaks" a bunch of tests etc.
+  const dsc: DynamicSamplingContext = {
     environment: options.environment || DEFAULT_ENVIRONMENT,
     release: options.release,
     public_key,
     trace_id,
-  }) satisfies DynamicSamplingContext;
+  };
+
+  if (!dsc.release) {
+    delete dsc.release;
+  }
+  if (!dsc.public_key) {
+    delete dsc.public_key;
+  }
```

Comment:
l: wondering when this is required tbh. Shouldn't we rather change our logic to treat `undefined` and "not at all defined" equally? Feel free to follow-up later though

m: also same question around public_key as in my other comment